src/mods/30-nvidia-safe-installer/install.sh

```
@@ -0,0 +1,196 @@
+    fi
+}
+
+# Check kernel compatibility for driver version
+check_kernel_compatibility() {
+    local kernel_version
+    kernel_version=$(uname -r)
+    # For simplicity, assume all drivers >= 450 support kernels >= 5.4
+    # This can be extended with a real compatibility matrix or ubuntu-drivers info
+    local recommended_driver_version=$1
+    local kernel_major=${kernel_version%%.*}
+    local kernel_minor=$(echo "$kernel_version" | cut -d. -f2)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)